### DISCUSSION TITLE
The engine get all data into memory one time or get data on different times during planning period

### DISCUSSION THREAD

**Discussion by sgfisme** [frePPLe/frepple#500] [Q&A]
Hi, 
I tried to figure out which deploy model fit on K8s. I deployed 2 model that one is separate pods for frePPLe and PostgreSQL and the other is that put the frePPLe and PostgreSQL on the one pod.
I use the same data set to test their performance, and the results are as below:

- frePPLe/PostgreSQL on one pods:
   - Data Initialization: 7.00 mins
   - Engine Execution: 41.51 mins
   - Export Plan: 113.58 mins
- frePPLe/PostgreSQL on different pods:
   - Data Initialization: 6.76 mins
   - Engine Execution: 54.7 mins
   - Export Plan: 132.9 mins
 
On export plan, it maybe has the network latency, but the time of plan generation on one pod is less than the deploy model on different pods. 

Just wondering to know that frePPLe get all of the planning metadata into the memory and then execute plan, or the engine gets data from PostgreSQL during the planning period?

**Comment by jdetaeye** [comment]
The engine execution part doesn't use the database. It's in memory calculations.  
(In the enterprise edition, it's a bit different. The forecast module reads forecast can read data "lazily" when it needs them.)

I would have expected the "engine execution" to have the same timing in both of your tests.
